pyk/src/pyk/cterm/cterm.py

```diff
@@ -334,7 +334,7 @@ def add_constraint(self, constraint: KInner) -> CSubst:
     def apply(self, cterm: CTerm) -> CTerm:
         """Apply this `CSubst` to the given `CTerm` (instantiating the free variables, and adding the constraints)."""
         _kast = self.subst(cterm.kast)
-        return CTerm(_kast, [self.constraint])
+        return CTerm(_kast, [self.constraint])  # <--
```

Comment:
On the other hand, when applying the `CSubst`, the `constraint` is used as-is.

Reply:
Hmm, wouldn’t it be better to maintain consistency? For example, using `self.constraints`? In fact, it’s not quite consistent within CTerm and CSubst. Ideally, constraints is a sugar for `#AND`. Thus,  I think that`#AND` shouldn’t exist in the elements of constraints, like [this](https://github.com/runtimeverification/k/blob/e6d4f896d7c51efd68f11c77141d457625472c43/pyk/src/pyk/cterm/cterm.py#L70). So we should add some operations to ensure this.

Another thing I think is that I may need a `constraint` property for CTerm to ensure consistency between CTerm and CSubst.